### DISCUSSION TITLE
How to contribute?

### DISCUSSION THREAD

**Discussion by henrywintif** [vrtmrz/obsidian-livesync#207] [Documentation requests]
I cloned the repository and opened it in PyCharm, where I have worked on TS projects before. It (the npm plugin) prompted me to install dependencies, so I did, and they were installed into a directory `node_modules` at the root of the project. When I look at `main.ts` it seems like dependencies are installed in `src/lib`.

A quickstart dev guide would be great so I can help contribute!

**Comment by vrtmrz** [comment]
I am delighted for your kindness!

For some time now, I have split some pieces of codes which are commonly used among LiveSync series (i.e. filesystem-livesync or livesync-classroom).
The common library is in https://github.com/vrtmrz/livesync-commonlib and registered as a submodule to this repository.

We have to clone this repo with the `--recursive` option, or if not, just fetch a submodule with the `git submodule update --init --recursive` command.
